src/tests.c

```diff
@@ -2854,8 +2894,10 @@ void run_sqrt(void) {
         for (j = 0; j < count; j++) {
             random_fe(&x);
             secp256k1_fe_sqr(&s, &x);
+            CHECK(secp256k1_fe_jacobi_var(&s) == 1);
             test_sqrt(&s, &x);
             secp256k1_fe_negate(&t, &s, 1);
+            CHECK(secp256k1_fe_jacobi_var(&t) == -1);
```

Comment:
Is it possible that that `secp256k1_fe_jacobi_var(&s)` is taking the slow `secp256k1_fe_sqrt` branch about 50% of the time? I discovered this on my machine with this simple patch that adds a printf:

```
diff --git a/src/field_5x52_impl.h b/src/field_5x52_impl.h
index 26e89123..d3eb3347 100644
--- a/src/field_5x52_impl.h
+++ b/src/field_5x52_impl.h
@@ -681,6 +681,7 @@ static int secp256k1_fe_jacobi_var(const secp256k1_fe *x) {
          * to computing a square root. This should be extremely rare with random
          * input. */
         secp256k1_fe dummy;
+        printf("slow");
         ret = 2*secp256k1_fe_sqrt(&dummy, &tmp) - 1;
 #ifdef VERIFY
     } else {
```

I'm quite confused because should be "extremely rare with random input", but `s` appears to be a random field element. Furthermore, "slow" isn't printed in the jacobi benchmarks unless I compile with `-DVERIFY` (which crashes after a few iterations, see other comment).

Reply:
Ah ok, that's good. I did not looked into `secp256k1_jacobi64_maybe_var` so far and hadn't considered that its behavior would change on VERIFY.